plugins/UFM_NDT_Plugin/ufm_sim_web_service/topo_diff/ndt_infra.py

```diff
@@ -453,11 +454,10 @@ def create_topoconfig_file(links_info_dict, ndt_file_path, patterns,
         error_message = "Topoconfig file creation failure: {} is empty or cannot be parsed: {}".format(ndt_file_path, te)
         logging.error(error_message)
         ndt_file.close()
-        return False, error_message
+        return False, error_message, failed_lable_conversion
     output_file = get_topoconfig_file_name(ndt_file_path) if not output_file_name else output_file_name
     # get mapping between node_guid and port lable to port number
     node_guid_lable2port_num = get_mapping_port_labels2port_numbers()
-    failed_lable_conversion = []
     with open(output_file, 'w') as topoconfig_file:
         for index, row in enumerate(dictreader):
             logging.debug("Parsing NDT link: {}".format(row))
```

Comment:
why do you call the parse_ndt_port twice (source & destination) for each NDT line instead of parsing the whole line at once?

Reply:
it is the same row - first time for source and second - for destination.
in this way it is implemented by Tolikin or Nahum